### ISSUE TITLE
Dereference errors in arrays of structs with bit-fields

### ISSUE THREAD

**Issue by shmarovfedor** [esbmc/esbmc#589]
I came across this issue recently:
```
#include <assert.h>

typedef struct str {
  unsigned short field1 : 4;
  unsigned short field2 : 4;
} str;

int main(void) {
  str array;
  str array_set[2];
  array_set[1] = array;
  assert(*(unsigned char *)&array == *(unsigned char *)&array_set[1]);
  return 0;
}
```
ESBMC returns `VERIFICATION FAILED` which should not be the case. However, when the size of each struct member is a multiple of 8 bits, the problem goes away.

**Comment by fbrausse** [comment]
I believe this is related to item 2. in https://github.com/esbmc/esbmc/pull/590#issuecomment-1010309733: pointers to structs seem to be interpreted as pointers to the struct's first member.

**Comment by fbrausse** [comment]
Fixed by #624.
